cmd/utils.go

```diff
@@ -95,9 +97,62 @@ func splitStr(path, sep string, n int) []string {
 	return splits
 }
 
+func buildS3ConfigFromEnv(envConfig, urlStr string) (*Config, *probe.Error) {
+	u, e := url.Parse(envConfig)
+	if e != nil {
+		return nil, probe.NewError(e).Trace(envConfig)
+	}
+	var (
+		accessKey    = u.User.Username()
+		secretKey, _ = u.User.Password()
+	)
+	if u.Scheme != "http" && u.Scheme != "https" {
+		return nil, errInvalidArgument().Trace(u.Scheme)
+	}
+
+	hostURL, e := url.Parse(u.Scheme + "://" + u.Host)
+	if e != nil {
+		return nil, probe.NewError(e).Trace(envConfig)
+	}
+
+	s3Config, err := newS3Config(cli.Args{hostURL.String(), accessKey, secretKey})
+	if err != nil {
+		return nil, err.Trace(envConfig)
+	}
+
+	s3Config.AppName = "mc"
+	s3Config.AppVersion = Version
+	s3Config.AppComments = []string{os.Args[0], runtime.GOOS, runtime.GOARCH}
+	s3Config.Debug = globalDebug
+	s3Config.Insecure = globalInsecure
+
+	return s3Config, nil
+}
+
+func getAliasFromEnv(envs []string) string {
+	for _, k := range envs {
+		if strings.HasPrefix(k, "MC_TARGET_") {
```

Comment:
Can you add a constant `const mcEnvPrefix = "MC_TARGET_"` ?

Reply:
:+1: